Tests/SubprocessTests/ProcessMonitoringTests.swift

```diff
@@ -0,0 +1,361 @@
+        return Configuration(
+            executable: .name("cmd.exe"),
+            arguments: ["/c", "exit \(code)"]
+        )
+        #else
+        return Configuration(
+            executable: .path("/bin/sh"),
+            arguments: ["-c", "exit \(code)"]
+        )
+        #endif
+    }
+
+    private func longRunningProcess(withTimeOutSeconds timeout: Double? = nil) -> Configuration {
+        #if os(Windows)
+        let waitTime = timeout ?? 99999
+        return Configuration(
+            executable: .name("powershell.exe"),
+            arguments: ["-Command", "Start-Sleep -Seconds \(waitTime)"]
+        )
+        #else
+        let waitTime = timeout != nil ? "\(timeout!)" : "infinite"
+        return Configuration(
+            executable: .path("/bin/sleep"),
+            arguments: [waitTime]
+        )
+        #endif
+    }
+
+    private func devNullInputPipe() throws -> CreatedPipe {
+        #if os(Windows)
+        let devnullFd: FileDescriptor = try .openDevNull(withAccessMode: .writeOnly)
+        let devnull = HANDLE(bitPattern: _get_osfhandle(devnullFd.rawValue))!
+        #else
+        let devnull: FileDescriptor = try .openDevNull(withAccessMode: .readOnly)
+        #endif
+        return CreatedPipe(
+            readFileDescriptor: .init(devnull, closeWhenDone: true),
+            writeFileDescriptor: nil
+        )
+    }
+
+    private func devNullOutputPipe() throws -> CreatedPipe {
+        #if os(Windows)
+        let devnullFd: FileDescriptor = try .openDevNull(withAccessMode: .writeOnly)
+        let devnull = HANDLE(bitPattern: _get_osfhandle(devnullFd.rawValue))!
+        #else
+        let devnull: FileDescriptor = try .openDevNull(withAccessMode: .writeOnly)
+        #endif
+        return CreatedPipe(
+            readFileDescriptor: nil,
+            writeFileDescriptor: .init(devnull, closeWhenDone: true)
+        )
+    }
+}
+
+// MARK: - Basic Functionality Tests
+extension SubprocessProcessMonitoringTests {
+    @Test func testNormalExit() async throws {
+        let config = self.immediateExitProcess(withExitCode: 0)
+        let spawnResult = try config.spawn(
+            withInput: self.devNullInputPipe(),
+            outputPipe: self.devNullOutputPipe(),
+            errorPipe: self.devNullOutputPipe()
+        )
+        defer {
+            spawnResult.execution.processIdentifier.close()
+        }
+        let monitorResult = try await monitorProcessTermination(
+            for: spawnResult.execution.processIdentifier
+        )
+
+        #expect(monitorResult.isSuccess)
+    }
+
+    @Test func testExitCode() async throws {
+        let config = self.immediateExitProcess(withExitCode: 42)
+        let spawnResult = try config.spawn(
+            withInput: self.devNullInputPipe(),
+            outputPipe: self.devNullOutputPipe(),
+            errorPipe: self.devNullOutputPipe()
+        )
+        defer {
+            spawnResult.execution.processIdentifier.close()
```

Comment:
Consider a withSpawnResult helper that guarantees closing the process descriptor? The pattern is repeated many times throughout the file and a future test author could forget to call close.